backend/routes/user.js

```diff
@@ -444,4 +443,59 @@ user.get("/budget/:userId", authenticate, async (req, res) => {
   }
 });
 
+// adds avoided merchant
+user.post("/avoided-merchant", authenticate, async (req, res) => {
+  const { merchantName } = req.body;
+  const userId = req.user.uid;
+
+  try {
+    const avoidedMerchant = await prisma.avoidedMerchant.create({
+      data: {
+        userId,
+        merchantName: merchantName.trim(),
+      },
+    });
+    res.json(avoidedMerchant);
+  } catch (error) {
+    if (error.code === "P2002") {
+      res.status(400).json({ error: "Merchant already in avoided list" });
+    } else {
+      res.status(500).json({ error: "Failed to add avoided merchant" });
+    }
+  }
+});
+
+// gets avoided merchants
+user.get("/avoided-merchants", authenticate, async (req, res) => {
+  const userId = req.user.uid;
+
+  try {
+    const avoidedMerchants = await prisma.avoidedMerchant.findMany({
+      where: { userId },
+      orderBy: { createdAt: "desc" },
+    });
+    res.json(avoidedMerchants);
+  } catch (error) {
+    res.status(500).json({ error: "Failed to fetch avoided merchants" });
+  }
+});
+
+// deletes avoided merchant
+user.delete("/avoided-merchant/:id", authenticate, async (req, res) => {
+  const { id } = req.params;
+  const userId = req.user.uid;
+
+  try {
+    await prisma.avoidedMerchant.delete({
+      where: {
+        id,
+        userId, // user can only delete personal entries
```

Comment:
should it be id only assume it's the primary key?
```suggestion
        id
```